crates/driver/src/domain/mempools.rs

```diff
@@ -112,9 +112,10 @@ impl Mempools {
         block_stream.next().await;
 
         let hash = mempool.submit(tx.clone(), settlement.gas, solver).await?;
+        let deadline = mempool.config().deadline();
```

Comment:
If we want to be super correct the deadline should be the first thing that gets computed (or at least before doing anything async).